src/module.c

```diff
@@ -6099,6 +6090,38 @@ robj **moduleCreateArgvFromUserFormat(const char *cmdname, const char *fmt, int
     return NULL;
 }
 
+/* Returns an array of robj pointers, by parsing the format specifier "fmt" as described for
+ * the RM_Call(), RM_Replicate() and other module APIs. Populates *argcp with the number of
+ * items (which equals to the length of the allocated argv).
+ *
+ * The integer pointed by 'flags' is populated with flags according
+ * to special modifiers in "fmt".
+ *
+ *     "!" -> REDISMODULE_ARGV_REPLICATE
+ *     "A" -> REDISMODULE_ARGV_NO_AOF
+ *     "R" -> REDISMODULE_ARGV_NO_REPLICAS
+ *     "3" -> REDISMODULE_ARGV_RESP_3
+ *     "0" -> REDISMODULE_ARGV_RESP_AUTO
+ *     "C" -> REDISMODULE_ARGV_RUN_AS_USER
+ *     "M" -> REDISMODULE_ARGV_RESPECT_DENY_OOM
+ *     "K" -> REDISMODULE_ARGV_ALLOW_BLOCK
+ *
+ * On error (format specifier error) NULL is returned and nothing is
+ * allocated. On success the argument vector is returned. */
+robj **moduleCreateArgvFromUserFormat(const char *cmdname, const char *fmt, int *argcp, int *flags, va_list ap) {
+    return moduleCreateArgvFromUserFormatAux(cmdname, fmt, argcp, flags, ap);
+}
+
+/* Returns an array of robj pointers, by parsing the format specifier "fmt" as described for
+ * the RM_SetAdd(), RM_SetRem() and other module APIs. Populates *argcp with the number of
+ * items (which equals to the length of the allocated argv).
+ *
+ * On error (format specifier error) NULL is returned and nothing is
+ * allocated. On success the argument vector is returned. */
+robj **moduleCreateSetArgvFromUserFormat(const char *fmt, int *argcp, va_list ap) {
```

Comment:
i'd rather not name it as specific for `Set`.
it's just a generic one that handles an array, in contrast to the other one that takes a command name.
ideally we would have renamed the other one to have `Cmd` in the function name.
another alternative (if we can't find a good name for this one) is to call moduleCreateArgvFromUserFormatAux directly (passing NULL).
not sure which one is better.

Reply:
Had same concern indeed. I'll change the name.